### PR TITLE
implemented the possibility to change the internalname of a file from…

### PR DESCRIPTION
**Pull request by Zwergal**
… the client

Please review

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#Tue Jan 19 01:57:51 CET 2016
+#Wed Jan 27 20:54:17 CET 2016
```
Comment (Narigo) [inline gradle/wrapper/gradle-wrapper.properties]: unnecessary

Comment (Zwergal) [inline gradle/wrapper/gradle-wrapper.properties]: How can I fix gradle to not set this automatically?

File: src/main/scala/com/campudus/tableaux/database/domain/file.scala [hunk 2]
```diff
@@ -57,14 +57,14 @@ case class ExtendedFile(file: File) extends DomainObject {
...
     val urls = file.externalName.values.map({
-      case (langtag, filename) =>
-        if (filename == null || filename.isEmpty) {
-          (langtag, null)
-        } else {
-          val encodedFilename = URLEncoder.encode(filename, "UTF-8")
-          (langtag, s"/files/${file.uuid.get}/$langtag/$encodedFilename")
-        }
-    })
+        case (langtag, filename) =>
```
Comment (Narigo) [inline src/main/scala/com/campudus/tableaux/database/domain/file.scala]: incorrect indendation?

File: src/main/scala/com/campudus/tableaux/controller/MediaController.scala [hunk 3]
```diff
@@ -146,8 +146,31 @@ class MediaController(override val config: TableauxConfig,
...
+        case Failure(ex) => p.failure(ex)
+        case _ => p.failure(UnknownServerException("Error in vertx filesystem exists check"))
+      }: Try[java.lang.Boolean] => Unit)
+
+      p.future
+    }
+
+    Future.sequence(internalName.values.map { x =>
+      val internalFileName = x._2
+      if (!internalFileName.split("[/\\\\]")(0).equals(internalFileName) ||
```
Comment (Narigo) [inline src/main/scala/com/campudus/tableaux/controller/MediaController.scala]: Why `[/\\\\]` and not only `[/\\]` ?
Would be good to add a test for `/something` and `\something` for example...

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Narigo) [inline gradle/wrapper/gradle-wrapper.properties]: I have no idea. :)

File: src/main/scala/com/campudus/tableaux/controller/MediaController.scala (hunk 3 above)
Comment (Zwergal) [inline src/main/scala/com/campudus/tableaux/controller/MediaController.scala]: I need 2 backslashes in the RegEx: RegEx should look like [/\\]. So the backslash in the RegEx is escaped and treated as the literal \. In the scala string I need to escape a backslash to be a literal \ so I need 4 backslashes :)

**Pull request merged**